### PR TITLE
Fix cp.push empty file

### PR DESCRIPTION
**Pull request by dincamihai**
Co-authored-by: Jochen Breuer <jbreuer@suse.de>

### What does this PR do?

When running `cp.push` I get this in salt-master log:

```
[ERROR   ] Error in function _file_recv:
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/salt/master.py", line 1776, in run_func
    ret = getattr(self, func)(load)
  File "/usr/lib/python3.6/site-packages/salt/master.py", line 1454, in _file_recv
    fp_.write(load['data'])
TypeError: a bytes-like object is required, not 'str'
```

Using `salt.utils.stringutils.to_str` didn't help so I am using plain `encode('utf8')`.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/master.py [hunk 1]
```diff
@@ -1452,7 +1452,7 @@ def _file_recv(self, load):
             if load['loc']:
                 fp_.seek(load['loc'])
 
-            fp_.write(load['data'])
+            fp_.write(load['data'].encode('utf8'))
```
Comment (terminalmage) [inline salt/master.py]: This needs to be changed to `salt.utils.stringutils.to_bytes(load['data'])`.

**Review by terminalmage — CHANGES_REQUESTED** [review]

File: salt/master.py (hunk 1 above)
Comment (dincamihai) [reply salt/master.py]: Thanks! Changing it now.

**Comment by rallytime** [comment]
re-run py3

**Review by terminalmage — APPROVED** [review]

**Comment by rallytime** [comment]
re-run py3

**Pull request merged**